mithril-common/src/chain_observer/model.rs

```diff
@@ -1,12 +1,42 @@
 use anyhow::anyhow;
-use serde::Serialize;
+use pallas_codec::minicbor::{decode, Decode, Decoder};
+use pallas_primitives::{conway::PlutusData, ToCanonicalJson};
+use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use std::collections::HashMap;
 use strum::{Display, EnumDiscriminants};
 use thiserror::Error;
 
 use crate::{StdError, StdResult};
 
+/// [Datum] represents an inline datum from UTxO.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+#[serde(rename_all = "lowercase")]
+pub struct Datum(pub PlutusData);
+
+impl<'a, C> Decode<'a, C> for Datum {
+    fn decode(d: &mut Decoder<'a>, ctx: &mut C) -> Result<Self, decode::Error> {
+        PlutusData::decode(d, ctx).map(Datum)
+    }
+}
+
+impl ToCanonicalJson for Datum {
+    fn to_json(&self) -> serde_json::Value {
+        self.0.to_json()
+    }
+}
+
+/// Inspects the given bytes and returns a decoded `R` instance.
+pub fn inspect<R>(inner: Vec<u8>) -> R
```

Comment:
Why didn't you implement the `TryFrom` trait here?

Alos, this should not panic, and return an error instead:
```suggestion
pub fn try_inspect<R>(inner: Vec<u8>) -> StdResult<R>
```